src/common/include/gudhi/graph_simplicial_complex.h

```diff
@@ -30,40 +31,127 @@ struct vertex_filtration_t {
+/** \brief Filtered_edges_vector contains the edges with their filtration values, sorted by ascending filtration values,
+ * in order to store the result of `Gudhi::compute_edge_graph` function.
  *
- * \tparam SimplicialComplexForProximityGraph furnishes `Filtration_value` type definition.
+ * \tparam SimplicialComplexForFilteredEdges furnishes `Filtration_value` and `Vertex_handle` types definitions.
  *
  */
-template <typename SimplicialComplexForProximityGraph>
-using Proximity_graph = typename boost::adjacency_list < boost::vecS, boost::vecS, boost::directedS
-, boost::property < vertex_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >
-, boost::property < edge_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >>;
+template <typename SimplicialComplexForFilteredEdges>
+class Filtered_edges_vector {
+ public:
+  /** \brief Type of the `Gudhi::Filtered_edges_vector` data structure.
+   */
+  using Filtered_edge = std::tuple<typename SimplicialComplexForFilteredEdges::Filtration_value,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle>;
+  using Filtered_edge_set = std::vector<Filtered_edge>;
+
+  template <class EdgeIterator, class EdgePropertyIterator>
+  Filtered_edges_vector(EdgeIterator first, EdgeIterator last, EdgePropertyIterator ep_iter,
+                        typename SimplicialComplexForFilteredEdges::Vertex_handle n) {
+    for (; first < last; ++first) {
+      edges_.push_back(std::make_tuple(*ep_iter, first->first, first->second));
+      ++ep_iter;
+    }
+    // By default the sort is done on the first element (Filtration_value) in ascending order.
+    std::sort(edges_.begin(), edges_.end());
+  }
 
-/** \brief Computes the proximity graph of the points.
+  Filtered_edge_set edges() const { return edges_; }
+
+  /** \brief Returns the edges vector at a specific index filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_at(std::size_t new_index) const {
+    return std::get<0>(*(edges_.begin() + new_index));
+  }
+
+  /** \brief Returns the edges vector min filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_min() const {
+    return std::get<0>(*(edges_.begin()));
+  }
+
+  /** \brief Returns the edges vector max filtration value.
+   */
+  typename SimplicialComplexForFilteredEdges::Filtration_value get_filtration_max() const {
+    return std::get<0>(*(edges_.rbegin()));
+  }
+
+  /** \brief Returns the edges vector.
+   */
+  std::size_t size() const { return edges_.size(); }
+
+  /** \brief Returns the sub-filtered edges vector from a `new_threshold` filtration value.
+   */
+  Filtered_edge_set sub_filter_edges_by_filtration(typename SimplicialComplexForFilteredEdges::Filtration_value new_threshold) const {
```

Comment:
`Filtered_edge_set` is not documented. Looking at what it is, it seems very similar to `Filtered_edges_vector`.
Do you really need to return a heavy vector? If the underlying `Filtered_edges_vector` doesn't disappear, a light `boost::iterator_range` looks like it might suffice.